### DISCUSSION TITLE
Allow custom modules

### DISCUSSION THREAD

**Discussion by Thomas-Philippot** [Jas-SinghFSU/HyprPanel#83] [Ideas]
I thought it would be nice to have custom modules like [waybar](https://github.com/Alexays/Waybar/wiki/Module:-Custom) does, to allow even more customization

**Comment by Jas-SinghFSU** [comment]
It's definitely on the roadmap!

**Comment by Jas-SinghFSU** [comment]
https://hyprpanel.com/configuration/modules.html#custom-modules

Added, just need to document how to contribute/create custom modules using the framework.

**Reply by Thomas-Philippot** [reply]
Nice, can we add any bash script ?

**Reply by Jas-SinghFSU** [reply]
Yep, you can make a simple bash script module or something more complex.

**Reply by byddha** [reply]
> https://hyprpanel.com/configuration/modules.html#custom-modules
> 
> Added, just need to document how to contribute/create custom modules using the framework.

Sorry in advance for bothering you about this - I've checked the docs multiple times and whatever relevant issues I could find, but I can't figure out if it's possible to build custom modules without modifying the source code directly (e.g for user specific modules that make no sense for any other user)

**Reply by Jas-SinghFSU** [reply]
No-code custom modules are definitely planned and I do have a vision for it as well. I just need to sit down and implement it. It'll let you make modules similarly to Waybar.

**Comment by Jas-SinghFSU** [comment]
@drzbida https://hyprpanel.com/configuration/custom_modules.html

Custom modules via JSON have been implemented. You can refer to the Wiki link above for info on how to make them.

**Reply by byddha** [reply]
Thank you very much! I've played with it quite a bit and didn't run into any issues so far, it works great! Maybe it's worth noting in the wiki that hyprpanel needs to be restarted for changes in `modules.json` to take effect (it took me a few minutes to figure out it's not dynamic like `config.json`), but that might be a me problem :)

**Reply by Thomas-Philippot** [reply]
Thanks for the update, I really appreciate ! 
Working great, but I had to try i logout / login for changes to take effect. 
At first I thought my json was wrong so yeah that might be a thing to add on the documentation.
